### PR TITLE
Improvements to streams usages

### PR DESCRIPTION
**Pull request by DomGarguilo**
Various improvements of current usage of streams.

### CODE REVIEW

File: core/src/test/java/org/apache/accumulo/core/conf/ConfigurationTypeHelperTest.java [hunk 1]
```diff
@@ -25,17 +25,17 @@
...
 import org.junit.jupiter.api.Test;
 
 public class ConfigurationTypeHelperTest {
 
   @Test
   public void testGetMemoryInBytes() {
-    Arrays.<Function<String,Long>>asList(ConfigurationTypeHelper::getFixedMemoryAsBytes,
-        ConfigurationTypeHelper::getMemoryAsBytes).stream().forEach(memFunc -> {
+    Stream.<Function<String,Long>>of(ConfigurationTypeHelper::getFixedMemoryAsBytes,
+        ConfigurationTypeHelper::getMemoryAsBytes).forEach(memFunc -> {
```
Comment (milleruntime) [inline core/src/test/java/org/apache/accumulo/core/conf/ConfigurationTypeHelperTest.java:38]: This one is neat. I haven't used it yet.

File: test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompactionMetricsIT.java [hunk 2]
```diff
@@ -166,7 +166,7 @@ public void testMetrics() throws Exception {
         UtilWaitThread.sleep(100);
         try (TabletsMetadata tm = getCluster().getServerContext().getAmple().readTablets()
             .forLevel(DataLevel.USER).fetch(ColumnType.ECOMP).build()) {
-          count = tm.stream().flatMap(t -> t.getExternalCompactions().keySet().stream()).count();
+          count = tm.stream().mapToLong(t -> t.getExternalCompactions().keySet().size()).sum();
```
Comment (milleruntime) [inline test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompactionMetricsIT.java:169]: I will have to look this data structure up again but I thought there was something unique about `flatMap`... Do you know if this translates to the same data structure?

File: test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_1_IT.java [hunk 3]
```diff
@@ -397,7 +397,7 @@ public void testExternalCompactionDeadTServer() throws Exception {
       LOG.info("Waiting for external compaction to complete.");
       TableId tid = getCluster().getServerContext().getTableId(table3);
       Stream<ExternalCompactionFinalState> fs = getFinalStatesForTable(getCluster(), tid);
-      while (fs.count() == 0) {
+      while (fs.findAny().isEmpty()) {
```
Comment (milleruntime) [inline test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_1_IT.java:400]: Cool!

File: test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompactionMetricsIT.java (hunk 2 above)
Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompactionMetricsIT.java:169]: The output of the user function that you pass to flatMap should be a Stream. So, you're converting the original stream into a stream of streams. flatMap then just concatenates them, so all the elements of each of those streams are put into one output stream.

These should be identical. Essentially, it goes from `sizeof(concat(set1, set2, ... setN))` to `sum(sizeof(set1), sizeof(set2), ... sizeof(setN))`.

File: test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_1_IT.java (hunk 3 above)
Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_1_IT.java:400]: This is a nice short-circuit.

**Review by ctubbsii — APPROVED** [review]
Overall, looks good. I do tend to prefer `Stream.of` over `Arrays.stream`, only because it's fewer characters, but either way works fine. Sometimes, you can also shorten some of the lines by statically importing the Collectors method, like `toSet` or `toList`, etc. But those are entirely optional, and only useful if the formatting makes it easier to read.

**Comment by DomGarguilo** [comment]
> I do tend to prefer `Stream.of` over `Arrays.stream`, only because it's fewer characters, but either way works fine

I did not know those two are equivalent. I think I like `Stream.of` better as well so I'll change them to that.

**Comment by DomGarguilo** [comment]
In [80eddf7](https://github.com/apache/accumulo/pull/2638/commits/80eddf7fc4270d0226ea3cdb0c8c72c0f23faf8b) I converted `Arrays.stream` to `Stream.of`. I also made some simplifications to PropertyTypeTest

**Pull request merged**